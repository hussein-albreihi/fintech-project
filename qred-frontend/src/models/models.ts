export interface Loan {
  id: number;
  userId: number;
  amount: number;
  interestRate: number;
  startDate: string;
  dueDate: string;
  status: 'ACTIVE' | 'PAID' | 'OVERDUE';
}

export interface User {
  id: number;
  name: string;
  email: string;
}

export interface UserWithLoans extends User {
  loans: Loan[];
}