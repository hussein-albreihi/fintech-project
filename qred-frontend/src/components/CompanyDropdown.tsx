// src/components/CompanyDropdown.tsx
import { useEffect, useState } from 'react';
import api from '../utils/api';
import { User } from '../models/models';

interface CompanyDropdownProps {
  onSelect: (user: User) => void;
  selectedUser: User | null;
}

const CompanyDropdown = ({ onSelect, selectedUser }: CompanyDropdownProps) => {
  const [users, setUsers] = useState<User[]>([]);
  const [isOpen, setIsOpen] = useState(false);

  useEffect(() => {
    api.get<User[]>('/users')
      .then((res) => setUsers(res.data))
      .catch(console.error);
  }, []);

  const handleSelect = (user: User) => {
    onSelect(user);
    setIsOpen(false);
  };

  return (
    <div className="relative">
      <button
        onClick={() => setIsOpen((prev) => !prev)}
        className="w-full bg-gray-100 rounded px-4 py-2 text-gray-800 font-medium flex justify-between items-center"
      >
        {selectedUser ? selectedUser.name : 'Select a company'}
        <span className="ml-2">âŒ„</span>
      </button>

      {isOpen && (
        <div className="absolute left-0 right-0 mt-1 bg-white border rounded shadow-md z-10">
          {users.map((user) => (
            <div
              key={user.id}
              className={`px-4 py-2 hover:bg-gray-100 cursor-pointer text-sm ${
                selectedUser?.id === user.id ? 'bg-gray-200 font-bold' : ''
              }`}
              onClick={() => handleSelect(user)}
            >
              {user.name}
            </div>
          ))}
        </div>
      )}
    </div>
  );
};

export default CompanyDropdown;
