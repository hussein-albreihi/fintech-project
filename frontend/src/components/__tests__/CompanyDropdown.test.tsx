import renderer from 'react-test-renderer';
import CompanyDropdown from '../CompanyDropdown';
import { User } from '../../models/models';

describe('CompanyDropdown Component', () => {
  it('renders correctly with no selected user', () => {
    const mockOnSelect = jest.fn();
    const tree = renderer
      .create(
        <CompanyDropdown onSelect={mockOnSelect} selectedUser={null} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a selected user', () => {
    const mockOnSelect = jest.fn();
    const mockUser: User = {
      id: 1, name: 'Test User',
      email: 'test@test.se'
    };
    const tree = renderer
      .create(
        <CompanyDropdown onSelect={mockOnSelect} selectedUser={mockUser} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});