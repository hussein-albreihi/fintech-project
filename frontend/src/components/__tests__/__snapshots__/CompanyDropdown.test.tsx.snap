// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyDropdown Component renders correctly with a selected user 1`] = `null`;

exports[`CompanyDropdown Component renders correctly with no selected user 1`] = `null`;
